capabilityreceiver/pkg/capabilityhandler/includedfields.go

```diff
@@ -88,3 +101,184 @@ func checkField(capInt map[string]interface{}, fieldNames []string) bool {
+			length := len(fieldNames)
+			return checkArrFieldExtension(fieldNames[nextIndex:length], fieldArr, url)
+		} else {
+			capInt = field.(map[string]interface{})
+		}
+	}
+
+	return false
+}
+
+// Given an array of interface objects, loops through each object to check whether the extension is populated following the path of fieldNames
+func checkArrFieldExtension(fieldNames []string, fieldArr []interface{}, url string) bool {
+	var found bool
+	// Loop through the array of interface objects
+	for _, obj := range fieldArr {
+		// For each object in interface array, get desired field using name in fieldNames
+		name := fieldNames[0]
+		objMap := obj.(map[string]interface{})
+		extensionField := objMap[name]
+		if extensionField == nil {
+			// If the desired field does not exist in that object, continue to the next object within the array of interface objects
+			continue
+		} else if name != "extension" && name != "modifierExtension" && arrContains(arrayFields, name) {
+			// If the desired field is not extension or modifierExtension and it is also an array of interface objects, call checkArrFieldExtension with this new array
+			fieldArr := extensionField.([]interface{})
+			length := len(fieldNames)
+			found = checkArrFieldExtension(fieldNames[1:length], fieldArr, url)
+			if found {
+				return found
+			}
+		} else if name != "extension" && name != "modifierExtension" && !arrContains(arrayFields, name) {
+			// If the desired field is not extension or modifierExtension and it is not an array of interface objects, call checkExtension with this field map[string]interface
+			extensionField := extensionField.(map[string]interface{})
+			length := len(fieldNames)
+			found = checkExtension(extensionField, fieldNames[1:length], url)
+			if found {
+				return found
+			}
+		} else {
+			// If the desired field is extension or modifierExtension, check array of extension interface objects for correct url
+			extensionArr := extensionField.([]interface{})
+			found = checkExtensionURL(extensionArr, url)
+			if found {
+				return found
+			}
+		}
+	}
+	return found
+}
+
+// Checks whether the given extension array contains the correct extension url
+func checkExtensionURL(extensionArr []interface{}, url string) bool {
+	found := false
+	for _, extension := range extensionArr {
+		extensionMap := extension.(map[string]interface{})
+		urlField := extensionMap["url"]
+		if urlField == url {
+			found = true
+			break
+		}
+	}
+	return found
+}
+
+// checkMultipleFieldsExtension loops through all the possible locations of the extension to see if it exists
+func checkMultipleFieldsExtension(capInt map[string]interface{}, url string, extensionString string) bool {
+	var extensionList [][]string
+	if extensionString == "capabilitystatement-prohibited" || extensionString == "conformance-prohibited" {
+		extensionList = [][]string{
+			{"rest", "resource", "interaction", "modifierExtension"},
+			{"rest", "resource", "searchParam", "modifierExtension"},
```

Comment:
Could you technically just use all of these paths as individual entries in the single extensionList structure?

Reply:
I was thinking no, and my reasoning for that is because you want it to return true if any one of these comes back as valid, so you need a way to differentiate it from all the other ones that will either be valid or not based on one location. If you disagree though I would be happy to talk about it and fix this!